freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaInstanceHealthDetailsService.java

```diff
@@ -0,0 +1,161 @@
+        return freeIpaClient.serverConnCheck(freeIpaClient.getHostname(), instance.getDiscoveryFQDN());
+    }
+
+    private RPCResponse<Boolean> toBooleanRpcResponse(RPCResponse<CheckResult> nodeHealth) {
+        RPCResponse<Boolean> response = new RPCResponse<>();
+        response.setSummary(nodeHealth.getSummary());
+        response.setResult(isHealthCheckPassing(nodeHealth));
+        response.setCount(nodeHealth.getCount());
+        response.setTruncated(nodeHealth.getTruncated());
+        response.setMessages(nodeHealth.getMessages());
+        response.setCompleted(nodeHealth.getCompleted());
+        response.setFailed(nodeHealth.getFailed());
+        response.setValue(nodeHealth.getValue());
+        return response;
+    }
+
+    private NodeHealthDetails parseMessages(RPCResponse<CheckResult> rpcResponse, InstanceMetaData instanceMetaData) {
+        NodeHealthDetails nodeResponse = new NodeHealthDetails();
+        nodeResponse.setName(instanceMetaData.getDiscoveryFQDN());
+        nodeResponse.setInstanceId(instanceMetaData.getInstanceId());
+        if (isHealthCheckPassing(rpcResponse)) {
+            nodeResponse.setStatus(InstanceStatus.CREATED);
+        } else {
+            nodeResponse.setStatus(InstanceStatus.UNHEALTHY);
+            nodeResponse.setIssues(rpcResponse.getMessages().stream().map(RPCMessage::getMessage).collect(Collectors.toList()));
+        }
+        return nodeResponse;
+    }
+
+    private NodeHealthDetails legacyParseMessages(RPCResponse<Boolean> rpcResponse, InstanceMetaData instanceMetaData) {
+        String precedingMessage = MESSAGE_UNAVAILABLE;
+        NodeHealthDetails nodeResponse = new NodeHealthDetails();
+        nodeResponse.setStatus(rpcResponse.getResult() ? InstanceStatus.CREATED : InstanceStatus.UNHEALTHY);
+        nodeResponse.setName(instanceMetaData.getDiscoveryFQDN());
+        nodeResponse.setInstanceId(instanceMetaData.getInstanceId());
+        boolean found = false;
+        for (RPCMessage message : rpcResponse.getMessages()) {
+            Matcher nodeMatcher = NEW_NODE_PATTERN.matcher(message.getMessage());
+            if (nodeMatcher.find()) {
+                found = true;
+            }
+            if (!found) {
+                LOGGER.info("No node for message: {}" + message.getMessage());
+            } else {
+                // When parsing the messages, if there's an error, the error
+                // appears in the preceding message.
+                if (EXTERNAL_COMMAND_OUTPUT.equals(message.getName())) {
+                    Matcher matcher = RESULT_PATTERN.matcher(message.getMessage());
+                    if (matcher.find()) {
+                        if (!STATUS_OK.equals(matcher.group(STATUS_GROUP))) {
+                            nodeResponse.addIssue(precedingMessage);
+                        }
+                    }
+                    precedingMessage = message.getMessage();
+                }
```

Comment:
precedingMessage = message.getMessage(); will have no effect here as mutating string will not change the string in nodeResponse above. Could append the message before addIssue()